### DISCUSSION TITLE
lit/html vs lit/static/html

### DISCUSSION THREAD

**Discussion by jvailius** [lit/lit#4656] [Q&A] [answered]
Practically I have an issue bundling some lit exports due to identical names for html and other exports from `lit/static-html.js`  vs `lit`. I was wondering to what extend they are interchangeable or if it would not be better to rename statics e.g. `html` from lit/static-html to  `htmlStatic`?

Then there is the `withStatic` with very little documentation. Is it the case that by using `withStatic`  I can avoid exporting/using `html` from ` lit/static-html.js `directly? If so, how?

**Comment by augustjk** [comment]
https://lit.dev/docs/templates/expressions/#static-expressions
Static versions of these tag functions are meant as sort of an escape hatch that allows expressions in what are normally static string portions of the template. They incur a performance overhead over the regular tag functions so should be used sparingly.

The "all" bundle made by the Lit team renames those to `staticHtml` and `staticSvg` as described here: https://lit.dev/docs/getting-started/#use-bundles

`withStatic()` is a helper function that are used to produce the static versions of the tag functions. Used like this: https://github.com/lit/lit/blob/abf30b3e895ea5d833f6d9559612e2b1ba47580d/packages/lit-html/src/static.ts#L172

**Reply by jvailius** [reply]
Thank you @augustjk,

> The "all" bundle made by the Lit team renames those to `staticHtml` and `staticSvg` as described here: https://lit.dev/docs/getting-started/#use-bundles

I am bundling lit myself and need to use the "default" imports from lit for various reasons so this is unfortunately not a solution.
 
> `withStatic()` is a helper function that are used to produce the static versions of the tag functions. Used like this:

Does it mean I can skip exporting html form static by using withStatic instead?

```js
import { html } from 'lit'
import { withStatic } from 'lit/static-html'

render() {
  return withStatic(html)`...some html ...`.
}
```

Then I just dont use the exports from 'lit/static-html' that have the same name.

**Reply by augustjk** [reply, marked as answer]
Yes, that would be fine.
